domain/storage/types.go

```diff
@@ -18,9 +20,9 @@ const (
 // Attrs defines storage attributes.
 type Attrs map[string]string
 
-// StoragePoolDetails defines the details of a storage pool to save.
-// This type is also used when returning query results from state.
-type StoragePoolDetails struct {
+// StoragePool represents a storage pool in Juju.
+// It contains the name of the pool, the provider type, and any attributes
+type StoragePool struct {
```

Comment:
One suggestion to consider with this is adding a member to this structure for the storage pool uuid. We know that where possible we want to start talking about things in terms of UUID. By starting to have it available allows for this transition to happen naturally.